crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+    /// var a = 1;
+    /// a = a * 1;
+    /// ```
+    ///
+    ///  ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a / 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a % 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a << 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a >> 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a >>> 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a & 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a ^ 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a | 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a ** 1;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a += 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a -= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a *= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a /= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a %= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a <<= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a &= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a ^= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a |= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a **= 1;
+    /// ```
+    ///
+    pub(crate) UseShorthandAssign {
+        version: "next",
+        name: "useShorthandAssign",
+        recommended: false,
+    }
+}
+
+pub struct RuleState {
+    shorthand_operator: JsSyntaxKind,
+    right: AnyJsExpression,
+}
+
+impl Rule for UseShorthandAssign {
+    type Query = Ast<JsAssignmentExpression>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let operator = node.operator().ok()?;
+        let right = node.right().ok();
+        let left = node.left().ok();
+        let left_var_name = left?.text();
+
+        if !matches!(operator, JsAssignmentOperator::Assign) {
+            return None;
+        }
+
+        let binary_expression = match right {
+            Some(AnyJsExpression::JsBinaryExpression(binary_expression)) => Some(binary_expression),
+            Some(AnyJsExpression::JsParenthesizedExpression(param)) => Some(
+                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?,
+            ),
+            _ => None,
```

Comment:
You can directly return `None`. This could allow dropping the `Some` in the previous results and enables using `binary_expression` instead of `binary_expression?`.
```suggestion
            Some(AnyJsExpression::JsBinaryExpression(binary_expression)) => binary_expression,
            Some(AnyJsExpression::JsParenthesizedExpression(param)) => 
                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?,
            _ => return None,
```